libafl/src/corpus/testcase.rs

```diff
@@ -123,8 +123,18 @@ where
 
     /// Set the filename
     #[inline]
-    pub fn set_filename(&mut self, filename: String) {
+    pub fn set_filename(&mut self, filename: String) -> Result<(), Error> {
+        let old_filename = self.filename.clone();
+
         self.filename = Some(filename);
+
+        match self.store_input() {
```

Comment:
If we store, do we want to remove the old version (if that exists)?

Reply:
To remove old file versions, we could implement a delete function for the Input trait.
But I'm wondering why this issue is even a problem on the first place? 
Isn't `set_filename` is only called when testcase.filename does not exist? Hence it is impossible to have self.filename with some string value.

Reply:
I think the solution here is to check if the file `.<filename>.lafl_lock` exists. If it does not, then it is safe to remove. Otherwise, return an illegal state error.

Reply:
I believe in this case, we also must not simply `store_input` but actually to _move_ the input file, the .metadata file, and the .lafl_lock (if each exist). Otherwise, this will simply create a copy.

Reply:
wouldn't we maybe have created the lock file, though?

Reply:
Yes, we would have made the lockfile. But even so, if we are renaming to something that *we currently use*, it would be unsafe to remove/replace the destination file.

Reply:
So what to do?

Reply:
I've added rename file logic. If either old lock file or new lock file exist, then set_filename will return error.

Reply:
Will that ever work, then? When is the lock file removed?

Reply:
I have added some more code to the functions.
So the idea is that old/new lock files are created and deleted in set_filename function. 

If lock file are already created for old/new  testcase then set_filename called, then it throws an error.
By doing this testcases could be safely created and removed.